test/server/admin/clusters_params_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "source/common/buffer/buffer_impl.h"
+#include "source/server/admin/clusters_params.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Server {
+
+struct ParamsCase {
+  std::string url_;
+  ClustersParams::Format expected_format_;
+  Http::Code expected_code_;
```

Comment:
one thing I'm wondering about with this pattern: will the test method names be reasonably understandable without looking at the code?

Note that an alternative is to make ParamsFixture method "testParams(...)" to run the common test, and explicitly create 3 named test-methods with the cases you want. It would be about the same number of lines, but I'm wondering if we'd bet more readable test method names.

Up to you; just throwing that out there.

Reply:
Right. When you parameterize the tests each parameter shows up in the test output by index. If you prefer, we can simply create 3 different tests and name them in a manner that is representative of the inputs. I'll make the adjustments.